### ISSUE TITLE
Implement a Forget Me feature

### ISSUE THREAD

**Issue by annahj1006** [ITU-BDSA23-GROUP4/Chirp#182]
## User story
When the user no longer want to be a part of Chirp, they need to be able to remove their information from our platform.

## Precise description of what to do and evt. time estimate
Implement a method that removes an author's cheeps, then removes who they are following and who are following them, and the lastly removes the author from the system. Making this a hard deletion
They should be logged out after the deletion has happendend, since we need the id upon deleting.

## Acceptance criteria (testeable)
There should be a red button on the buttom on the UserPage
The button should give an alert box - informing the user that their information will be deleted, giving them the option to change their mind
When deleting, they should then be logged out and redirected to the public page

Then test the created method in unit tests
- [x] That an author is deleted
- [x] That an authors cheeps is deleted
- [x] That the authors following list is emtied
- [x] That the author dissapears from other authors following lists

Dependent on issue #166

**Comment by annahj1006** [comment]
Cannot be completed until following works

**Comment by annahj1006** [comment]
Is not closed with pull request 227, since the follow/unfollow is missing, as is said in the pull request

**Comment by Lukan707** [comment]
### BUG ###

The already merged feature, that an authors cheeps is deleted, does not work.
The bug occurs both on a localhost instance of main and on production (Azure).

See issue #249 for reproduction of bug

**Comment by Lukan707** [comment]
> ### BUG
> The already merged feature, that an authors cheeps is deleted, does not work. The bug occurs both on a localhost instance of main and on production (Azure).
> 
> See issue #249 for reproduction of bug

This should be resolved when pull request #252 is merged into main.

**Comment by Lukan707** [comment]
### BUG ###

The user is not logged out when pressing "Forget Me".

See issue #253 for reproduction of bug

**Comment by Lukan707** [comment]
> ### BUG
> The user is not logged out when pressing "Forget Me".
> 
> See issue #253 for reproduction of bug

This is resolved with pull request #255, which is merged into main

**Comment by Lukan707** [comment]
The last part of the feature, deletion of followers relationship, is already functioning, because the table has deletion cascading from author.

There is also unit tests, that tests the acceptance requirements.
